### PR TITLE
feat: add response cache profile by country

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Useful to cache different settings for showing the cookie banner depending on user country

Related to https://app.clickup.com/t/2mn0q9a and https://app.clickup.com/t/2k5wph4

Dependency of https://github.com/ArdentHQ/msq.io/pull/285

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: src/Support/ResponseCache/ResponseCacheProfileByCountry.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Support\Facades\Auth;
+use Spatie\ResponseCache\CacheProfiles\CacheAllSuccessfulGetRequests;
+
+final class ResponseCacheProfileByCountry extends CacheAllSuccessfulGetRequests
+{
+    public function useCacheNameSuffix(Request $request): string
+    {
+        if (Auth::check()) {
+            return (string) Auth::id();
+        }
```
Comment (ItsANameToo) [inline src/Support/ResponseCache/ResponseCacheProfileByCountry.php:18]: do we need this as part of `ResponseCacheProfileByCountry`? 🤔 this is not related to country of origin and would also results in separate caches for each logged in user if registrations would be made open to the public at some point. Would be better as a separate class or to leave it for now until we run into a scenario where we need separate caches per user

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Comment by ItsANameToo** [comment]
also conflicts @alfonsobries

File: src/Support/ResponseCache/ResponseCacheProfileByCountry.php (hunk 1 above)
Comment (alfonsobries) [reply src/Support/ResponseCache/ResponseCacheProfileByCountry.php:18]: @ItsANameToo the default profile already separate by user (see the class that I'm extending) and the public site is different for logged users because they can see draft articles, I think that all the logged users see the exact same public site (since las time I checked draft articles are shown for All logged users not depending on the user/role for example)

So we can separate by loggenin+country but I think that using the id+country is better, is not like we are going to have hundred of users logged at the same time and that will make it this class more reusable on other projects 

What do you think?

Comment (ItsANameToo) [reply src/Support/ResponseCache/ResponseCacheProfileByCountry.php:18]: ah I see 🤔 fair enough for the auth thing then.

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
only conflict to resolve then @alfonsobries

**Comment by alfonsobries** [comment]
@ItsANameToo updated ✅

**Review by ItsANameToo — APPROVED** [review]
